iina/MPVController.swift

```diff
@@ -760,20 +774,53 @@ class MPVController: NSObject {
     }
   }
 
+  /// Tell Cocoa to terminate the application.
+  ///
+  /// - Note: This code must be in a method that can be a target of a selector in order to support macOS 10.11.
+  ///     The `perform` method in `RunLoop` that accepts a closure was introduced in macOS 10.12. If IINA drops
+  ///     support for 10.11 then the code in this method can be moved to the closure in `handleEvent and this
+  ///     method can then be removed.`
+  @objc
+  internal func terminateApplication() {
+    NSApp.terminate(nil)
+  }
+
   // Handle the event
   private func handleEvent(_ event: UnsafePointer<mpv_event>!) {
     let eventId = event.pointee.event_id
 
     switch eventId {
     case MPV_EVENT_SHUTDOWN:
-      let quitByMPV = !player.isMpvTerminated
+      let quitByMPV = !player.isShuttingDown
       if quitByMPV {
+        // This happens when the user presses "q" in a player window and the quit command is sent
+        // directly to mpv. The user could also use mpv's IPC interface to send the quit command to
+        // mpv. Must not attempt to change a mpv setting in response to an IINA preference change
+        // now that mpv has shut down. This is not needed when IINA sends the quit command to mpv
+        // as in that case the observers are removed before the quit command is sent.
+        removeOptionObservers()
+        // Submit the following task synchronously to ensure it is done before application
+        // termination is started.
         DispatchQueue.main.sync {
-          NSApp.terminate(nil)
+          self.player.mpvHasShutdown(isMPVInitiated: true)
+        }
+        // Initiate application termination. AppKit requires this be done from the main thread,
+        // however the main dispatch queue must not be used to avoid blocking the queue as per
+        // instructions from Apple.
+        if #available(macOS 10.12, *) {
```

Comment:
Terminating one mpv instance should not cause the whole app to be terminated. I know this was the behavior before, but I think it's a good time to correct that behavior.